### PR TITLE
feat: ✨ add sd-media + sd-copyright - CSS style

### LINKED ISSUE #276: feat: ✨ add sd-media + sd-copyright - CSS style
## User Story
As a user of the Solid Design System, I would like to utilize a style component to preview an image or a video with an optional caption and copyright, so that media content can be displayed semantically correct and accessible.
## Documentation
https://www.figma.com/file/EYUUCI9hFhfxKB5zn3jpju/Media?type=design&node-id=1705-4634&mode=design&t=HA8PYxdKVbxgblUJ-4

### Copyright

CleanShot 2024-02-09 at 11 07 32

https://www.figma.com/file/8u2hIRmhVX3mjW8462Au5I/Image?type=design&node-id=1705-4634&mode=design&t=hDEWaIwPrHkdNn0M-11

### Description

CleanShot 2024-02-09 at 11 08 31

>Note: 
The Figma component needs to be adjusted as well => not blocking and can be done in parallel 

## Technical Hint

To force best semantics, we will utilize figure and figcaption:

```html
<figure class="sd-media">
      <img src="https://image.url">
    <figcaption>
      This is a media caption
    </figcaption>
  </figure>
```

https://codepen.io/mariohamann/pen/LYaJJJN

For ease of use, we will use a CSS custom property for copyrights:

```html
<div class="sd-copyright" style="--copyright: '© 2024 Solid Design System';">
      <img src="https://image.url">
</div>
```

Combined this should result in the following prototype:

CleanShot 2024-02-09 at 11 18 54

https://codepen.io/mariohamann/pen/LYaJJJN

Have in mind, that it has to be ensured, that the copyright has to be able to be multilined depending on the width of its container.

## Hand-off
https://www.figma.com/file/EYUUCI9hFhfxKB5zn3jpju/Media?type=design&node-id=1503-4722&mode=design&t=HA8PYxdKVbxgblUJ-4

#### Props
| Status | Name | Type | Default | Description |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| <ul><li>[ ] </li></ul>  | `--copyright` | `CSS Variable` | `false` | displaying the copyright (always in the bottom left corner |

#### Stories (besides the `default` story)
  - [ ] `Copyright and caption` - showing both additional content sections

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [x] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by saemik94**
## Description:
closes #276 

## Definition of Reviewable:
*PR notes: Irrelevant elements should be removed.*
- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated
- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/styles/copyright/copyright.stories.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export const Default = {
+  parameters: {
+    controls: {
+      disable: true
+    }
+  },
+  render: (args: { [k: string]: any }) => {
+    return generateTemplate({
+      options: {
+        templateContent: `<div class="%CLASSES% max-w-xl" style="--copyright: '${args['--copyright']}';">%SLOT%</div>`
```
Comment (saemik94) [inline packages/components/src/styles/copyright/copyright.stories.ts:40]: Do I use the args the right way here, or how would you allow the --copyright property to be edited in storybook?

File: packages/components/src/styles/media/media.stories.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  },
+  render: (args: any) => {
+    return generateTemplate({
+      options: {
+        templateContent: `
+          <figure class="%CLASSES% max-w-xl p-4">
+            <div class="sd-copyright" style="--copyright: '© 2024 Solid Design System';">
+              <img src="./placeholders/generic.jpg" alt="A generic placeholder jpg" />
+            </div>
+            <figcaption>%SLOT%</figcaption>
```
Comment (saemik94) [inline packages/components/src/styles/media/media.stories.ts:52]: Is it ok to use the Slot Placeholder in order to make the figcaption editable in storybook?

**Comment by karlbaumhauer** [comment]
@saemik94 can we remove the skip chromatic from the PR? this prevents storybook from publishing it and therewith I cannot review it visually. Or what was the intention to use the suffix?

**Comment by saemik94** [comment]
@karlbaumhauer just wanted to wait until my questions have been answered, in case I have to make adjustements.

**Comment by yoezlem** [comment]
Should the sd-media only use in combination with images? Had in mind, that we wanted to use that with video and image

**Comment by saemik94** [comment]
@yoezlem I guess your question came up due to this part:

image

I implemented it, as it is defined in the technical hint and also in the codepen from @mariohamann. I guess you can wrap all kind of media inside the <figure> - or am I wrong or even got your question wrong?

File: packages/components/src/styles/copyright/copyright.stories.ts (hunk 1 above)
Comment (mariohamann) [reply packages/components/src/styles/copyright/copyright.stories.ts:40]: If it works, then it should be fine!

**Comment by mariohamann** [comment]
> @karlbaumhauer just wanted to wait until my questions have been answered, in case I have to make adjustements.

You can make a draft PR then.

**Comment by mariohamann** [comment]
> @yoezlem I guess your question came up due to this part:
> 
> image
> 
> I implemented it, as it is defined in the technical hint and also in the codepen from @mariohamann. I guess you can wrap all kind of media inside the
> 
> - or am I wrong or even got your question wrong?

That's the correct.

File: packages/components/src/styles/copyright/copyright.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.sd-copyright {
+  @apply relative;
+
+  &::after {
+    @apply block absolute p-4 bottom-0 left-0 text-white font-bold w-full content-[--copyright];
+    text-shadow: 1px 1px 2px black;
```
Comment (mariohamann) [inline packages/components/src/styles/copyright/copyright.css:6]: Did you check this in design?

**Review by mariohamann — APPROVED** [review]
https://github.com/solid-design-system/solid/pull/755#discussion_r1495784807

If this is fine for design, I'm fine, too.:)

File: packages/components/src/styles/copyright/copyright.css (hunk 3 above)
Comment (saemik94) [reply packages/components/src/styles/copyright/copyright.css:6]: oh very good question, this is a copy from your codePen sample. I just saw in Figma it's: `box-shadow: 0px 1px 3px 0px #515151BF;` but the color seems to be a manually set one. @solid-design-system/core-design which one should I use?

**Comment by coraliefeil** [comment]
Pls reassign for another review

**Review by yoezlem — APPROVED** [review]

**Comment by coraliefeil** [comment]
Approved. THX!

**Pull request merged**